StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -1233,34 +1233,35 @@ StETofCalibMaker::processStEvent()
     // collect status bit information and fill good event flag for 2020+ data
 	 TClass* headerClass = etofHeader->IsA();
 	 if( headerClass->GetClassVersion() > 2 ){
-	 	mNStatusBitsCounter.clear();			
-	  std::vector< Bool_t >  vMissmatchVec = etofHeader->missMatchFlagVec();
-		int iGet4Id = 0;
-		for( auto iMissMatchFlag : vMissmatchVec ){
-			// From DigiMaker:
-		   // mMissMatchFlagVec.at( 144 * ( sector - 13 ) + 48 * ( zplane -1 ) + 16 * ( counter - 1 ) + 8 * ( side - 1 ) + ( ( strip - 1 ) / 4 ) ) = true;
-		   if (iMissMatchFlag == false) continue;			
-			int iCounter = iGet4Id / 16;
-			if( mNStatusBitsCounter.count(iCounter) ){
-				mNStatusBitsCounter[iCounter]++;
-			}else{
-				mNStatusBitsCounter[iCounter] = 1;
-			}	
-		 }
 		
-		std::vector<bool> goodEventFlagVec; 	
-	  	for( int iCounter = 0; iCounter < 108; iCounter++){
-	  		if ( !(mNPulsersCounter.count(iCounter) ) ){
-	  			goodEventFlagVec.push_back(false);
-	  		}else{
-		  		if ( !(mNStatusBitsCounter.count(iCounter)) && mNPulsersCounter[iCounter] == 2){
-		  			goodEventFlagVec.push_back(true); //true when 2 pulser digis and zero status bits are available on this counter
-		  		}else{
-		  			goodEventFlagVec.push_back(false);
-		  		}
-		  	}
+		std::vector<bool> goodEventFlagVec; 
+		std::vector<bool> hasPulsersVec;
+		
+		//drag along pulser information
+		for( int iCounter = 0; iCounter < 108; iCounter++){
+		  if ( !(mNPulsersCounter.count(iCounter) ) ){
+		    hasPulsersVec.push_back(false);
+		  }else{
+		    if(mNPulsersCounter[iCounter] == 2){
+		      hasPulsersVec.push_back(true);
+		    }else{
+		      hasPulsersVec.push_back(false);
+		    }		    
+		  }		  
+		}
+		if (hasPulsersVec.size() == 108){
+		  //etofHeader->setHasPulsersVec(hasPulsersVec);  // not working but not of relevance at the moment 
+	  	} 
+
+		//fill good event flag into header
+		for( int iGet4 = 0; iGet4 < 1728; iGet4++){	  	
+		  if ( etofHeader->missMatchFlagVec().at(iGet4)){
+		    goodEventFlagVec.push_back(false); 
+		  }else{
+		    goodEventFlagVec.push_back(true);
+		  }
```

Comment:
This should be re-written in modern C++ without using an explicit loop. The both source and destination vectors appear to contain boolean values, so using std::copy with back_inserter should work.